### PR TITLE
Don't communicate the SuiteChannelManager through a zone

### PR DESCRIPTION
**Pull request by natebosch**

### CODE REVIEW

**Comment by natebosch** [comment]
This one isn't super critical since it already wraps all of `main()` from the user code so it isn't impacted by changes to run test bodies in the declaring zone.

File: pkgs/test_api/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## 0.3.0-dev
```
[the remark this answers is not in the document]
Comment (natebosch) [reply pkgs/test_api/CHANGELOG.md:1]: I'd guess we probably have some `src/` imports, I haven't looked closely yet at how this impacts internal use.

File: pkgs/test_core/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## 0.4.0-dev
+
+* Remove `suiteChannel` and `setStackTraceMapper` utilities. These are now
```
[the remark this answers is not in the document]
Comment (natebosch) [reply pkgs/test_core/CHANGELOG.md:3]: Ah yeah, I was going to remove that one too but hit some road blocks.

File: pkgs/test_api/CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (natebosch) [reply pkgs/test_api/CHANGELOG.md]: We can roll this in with another breaking change.

**Review by grouma — APPROVED** [review]

**Comment by natebosch** [comment]
I think this is worth landing to drop one level of zones. I gave another try at dropping the stack trace formatter part of it and it's a little bit more work, but I don't want that to block this easy win.

I changed it from exposing the `SuiteChannelManager` entirely to exposing only the callback that the use cases need. Will see if this has any negative impact internally.

@jonahwilliams - I found a reference to this in `flutter_tools` that could be a problem, but I don't understand it.

https://github.com/flutter/flutter/blob/891511d58f6550ce9e9b03b8d7c6a602caa97488/packages/flutter_tools/lib/src/web/bootstrap.dart#L139-L145

What is that accomplishing? It isn't used locally within that generated file, and nothing should be importing that to use it right? Can that be removed?

**Comment by jonahwilliams** [comment]
@natebosch I believe that code is unused now. I will try to remove it and report back

**Comment by jonahwilliams** [comment]
Seems like we can remove that, I linked that PR that will do so

**Pull request merged**